beacon-chain/p2p/handshake.go

```diff
@@ -22,7 +22,57 @@ const (
 )
 
 func peerMultiaddrString(conn network.Conn) string {
-	return fmt.Sprintf("%s/p2p/%s", conn.RemoteMultiaddr().String(), conn.RemotePeer().String())
+	remoteMultiaddr := conn.RemoteMultiaddr().String()
+	remotePeerID := conn.RemotePeer().String()
+	return fmt.Sprintf("%s/p2p/%s", remoteMultiaddr, remotePeerID)
+}
+
+func (s *Service) connectToPeer(conn network.Conn) {
+	s.peers.SetConnectionState(conn.RemotePeer(), peers.PeerConnected)
+	// Go through the handshake process.
+	log.WithFields(logrus.Fields{
+		"direction":   conn.Stat().Direction.String(),
+		"multiAddr":   peerMultiaddrString(conn),
+		"activePeers": len(s.peers.Active()),
+	}).Debug("Initiate peer connection")
+}
+
+func (s *Service) disconnectFromPeer(
```

Comment:
The naming here I find a tad confusing, it seems we only use this function to disconnect from a peer when we have an error, not a general disconnection. Would it be clearer to have it named something like `disconnectPeerOnError`?

Reply:
Fixed in 868f6838e3e4e448522ff733007cecd956e6e007.

Reply:
Fixed in 95fdbb4602e21141f804f61133feb00ebc260b09.